java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -458,7 +458,7 @@ public String toString(int index, int length, Charset charset) {
       return "";
     }
 
-    return ByteBufUtil.decodeString(this, index, length, charset);
+    return super.toString(index, length, charset);
```

Comment:
Yes. super.toString(index, length, charset); calls  ByteBufUtil.decodeString(this, index, length, charset)